modeling/main.py

```diff
@@ -0,0 +1,282 @@
+grp_name = 'singlechannel-training'
+
+parser = argparse.ArgumentParser(description='Script for training custom models for SCI Lesion Segmentation.')
+# Arguments for model, data, and training and saving
+parser.add_argument('-e', '--only_eval', default=False, action='store_true', help='Only do evaluation, i.e. skip training!')
+parser.add_argument('-m', '--model_type', choices=['unet'], default='unet', type=str, help='Model type to be used')
+# dataset
+parser.add_argument('-dr', '--dataset_root', 
+                    default='/home/GRAMES.POLYMTL.CA/u114716/duke/temp/muena/sci-zurich_preprocessed', type=str,
+                    help='Root path to the BIDS- and ivadomed-compatible dataset')
+parser.add_argument('-fd', '--fraction_data', default=1.0, type=float, help='Fraction of data to use. Helps with debugging.')
+parser.add_argument('-fho', '--fraction_hold_out', default=0.2, type=float, help='Fraction of data to hold-out of for the test phase')
+parser.add_argument('-ftv', '--fraction_train_val', nargs='+', default=[0.65, 0.15], 
+                    help="Train and validation split. Should sum to (fraction_data - fraction_hold_out)")
+# model 
+parser.add_argument('-t', '--task', choices=['sc', 'mc'], default='sc', type=str, help="Single-channel or Multi-channel model ")
+parser.add_argument('-bnf', '--base_n_filter', default=8, type=int, help="Number of Base Filters")
+parser.add_argument('-dep', '--depth', default=4, type=int, help="Depth of Simple UNet3D")
+parser.add_argument('-ccs', '--center_crop_size', nargs='+', default=[128, 256, 96], help='List containing center crop size for preprocessing')
+parser.add_argument('-svs', '--subvolume_size', nargs='+', default=[128, 256, 96], help='List containing subvolume size')
+parser.add_argument('-srs', '--stride_size', nargs='+', default=[128, 256, 96], help='List containing stride size')
+# optimizations
+parser.add_argument('-p', '--precision', default=32, type=int, help='Precision for training')
+parser.add_argument('-ne', '--num_epochs', default=500, type=int, help='Number of epochs for the training process')
+parser.add_argument('-bs', '--batch_size', default=2, type=int, help='Batch size of the training and validation processes')
+parser.add_argument('-nw', '--num_workers', default=4, type=int, help='Number of workers for the dataloaders')
+parser.add_argument('-lr', '--learning_rate', default=1e-4, type=float, help='Learning rate for training the model')
+parser.add_argument('-wd', '--weight_decay', type=float, default=0.01, help='Weight decay (i.e. regularization) value in AdamW')
+parser.add_argument('-pat', '--patience', default=250, type=int, help='number of validation steps (val_every_n_iters) to wait before early stopping')
+parser.add_argument('--T_0', default=100, type=int, help='number of steps in each cosine cycle')
+parser.add_argument('-epb', '--enable_progress_bar', default=False, type=bool, help='by default is disabled since it doesnt work in colab')
```

Comment:
I would set this to `True` by default. Otherwise, a newcomer to the script might wonder why no progress bar is showing up.